src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettings.tsx

```diff
@@ -66,11 +71,11 @@ const UserSecureSettings: React.FC<SecureSettingsProps> = ({ currentSettings })
     <PageBlock
       title="User Secure Settings"
       headerRight={
-        !edit ? (
+        !edit && !isGuest(user.role) ? (
```

Comment:
you should be able to do one isGuest check for this whole statement: `!isGuest(user.role) && (is edit ? :)`
The && returns the right side of the operator only if the left side is true